src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingXmlDocumentConstructedWithNoSecureResolverTests.cs

```diff
@@ -0,0 +1,874 @@
+        }
+
+        [Fact]
+        public void XmlDocumentAsFieldSetResolverToNullInSomeMethodShouldGenerateDiagnostics()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        public XmlDocument doc = new XmlDocument();
+
+        public void Method1()
+        {
+            this.doc.XmlResolver = null;
+        }
+
+        public void Method2(XmlReader reader)
+        {
+            this.doc.Load(reader);
+        }
+    }
+}",
+                GetCA3075XmlDocumentWithNoSecureResolverCSharpResultAt(8, 28, "TestClass")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+        Public doc As XmlDocument = New XmlDocument()
+
+        Public Sub Method1()
+            Me.doc.XmlResolver = Nothing
+        End Sub
+
+        Public Sub Method2(reader As XmlReader)
+            Me.doc.Load(reader)
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XmlDocumentWithNoSecureResolverBasicResultAt(6, 16, "TestClass")
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentCreatedAsTempNotSetResolverShouldGenerateDiagnostics()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+
+        public void Method1()
+        {
+            Method2(new XmlDocument());
+        }
+
+        public void Method2(XmlDocument doc){}
+    }
+}",
+                GetCA3075XmlDocumentWithNoSecureResolverCSharpResultAt(11, 21, "Method1")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+
+        Public Sub Method1()
+            Method2(New XmlDocument())
+        End Sub
+
+        Public Sub Method2(doc As XmlDocument)
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XmlDocumentWithNoSecureResolverBasicResultAt(8, 21, "Method1")
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentDerivedTypeNotSetResolverShouldNotGenerateDiagnostics()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass1 : XmlDocument
+    {
+        public TestClass1()
+        {
+            XmlResolver = null;
+        }
+    }     
+
+    class TestClass2
+    {
+        void TestMethod()
+        {
+            var c = new TestClass1();
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass1
+        Inherits XmlDocument
+        Public Sub New()
+            XmlResolver = Nothing
+        End Sub
+    End Class
+
+    Class TestClass2
+        Private Sub TestMethod()
+            Dim c = New TestClass1()
+        End Sub
+    End Class
+End Namespace
+"
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentDerivedTypeWithNoSecureResolverShouldNotGenerateDiagnostic()
+        {
```

Comment:
That's fine, but the worth a comment.

Sent from my Windows Phone
________________________________
From: Rick Qing Xu<mailto:notifications@github.com>
Sent: ‎2/‎2/‎2016 7:29 PM
To: dotnet/roslyn-analyzers<mailto:roslyn-analyzers@noreply.github.com>
Cc: Larry Golding<mailto:Laurence.Golding@microsoft.com>
Subject: Re: [roslyn-analyzers] Add XML rules CA3075 Only (#705)


In src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingXmlDocumentConstructedWithNoSecureResolverTests.cs<https://github.com/dotnet/roslyn-analyzers/pull/705#discussion_r51674617>:

> +        End Sub
> +    End Class
> +
> +    Class TestClass2
> +        Private Sub TestMethod()
> +            Dim c = New TestClass1()
> +        End Sub
> +    End Class
> +End Namespace
> +"
> +            );
> +        }
> +
> +        [Fact]
> +        public void XmlDocumentDerivedTypeWithNoSecureResolverShouldNotGenerateDiagnostic()
> +        {


This is desired. This actually verifies 3075 shouldn't check anything done in 3077. 3077 will report on code class DerivedType : XmlDocument {}

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn-analyzers/pull/705/files#r51674617>.